src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -890,6 +890,14 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
                 return op1;
             }
 
+#if defined(TARGET_X86)
+            if ((intrinsic == NI_VectorT_ToScalar) && varTypeIsLong(simdBaseType) &&
+                !compOpportunisticallyDependsOn(InstructionSet_SSE41))
+            {
+                return nullptr;
+            }
```

Comment:
nit: Most similar checks are higher up in a switch case before we ever get to this point

It's not necessarily important that we move that up there however, especially since the rest of this file should be getting removed in early .NET 10 and so there isn't any long term risk of breakage from other code or refactorings. I'll leave it up to you on if you want to just keep it "as is" or move it up with the rest of the handling.


-- For context, this file is the last of the `SimdAsHWIntrinsic` logic which now that Vector2/3/4 are in managed only exists to support `Vector<T>`. The plan for .NET 10 is to remove it and just forward by name to the Vector128/256/512 paths instead so we can reduce the cost and risk of bugs like the one this is fixing.

Reply:
Ok I will move it up.